### DISCUSSION TITLE
torchtransformers R package

### DISCUSSION THREAD

**Discussion by farach** [farach/huggingfaceR#8] [General]
Hey @samterfa! This might be a useful R package to incorporate into ours: https://github.com/macmillancontentscience/torchtransformers

> The goal of torchtransformers is to facilitate working with transformer-based models in R using torch.
> 
> This package is under active development, and the API might change before we reach our first CRAN release. Please check the issues to see if anything we have planned will impact your use cases.

What do you guys think? They have a couple of vignettes that are helpful to understand what they are doing in the package.

**Comment by jpcompartir** [comment]
Hey, finally going to get up to speed here I think - torchtransformers is made by Jonthegeek @ R4DS, could be worth reaching out to him directly.  Speaking of vignettes, could probably move most of the things from the current readme into a vignette.

**Reply by farach** [reply]
So glad to see you here @jpcompartir! What is a good email address to contact you? @samterfa and I would like to put together a Google doc to begin organizing a roadmap to a first release. Having a vignette as part of that would be great!

**Comment by R-icntay** [comment]
Hello @farach, @jpcompartir, @samterfa firstly thanks a lot for getting ahead with this. I asked whether anyone was working on a Huggingface transformers package in R a few weeks ago on twitter and really pumped this is coming to life.

Happy to contribute to the vignettes too! Any roadmap in place that could serve as a guide?

Thank you.

**Reply by jpcompartir** [reply]
Hey, we're looking at putting a roadmap together - will look to provide an update ASAP!

**Reply by DataStrategist** [reply]
I would love to see vignettes! I'd be happy to contribute by installing packagedown on this so there can be a nice website. Would you be interested in that?

**Reply by jpcompartir** [reply]
Hey, sorry I missed this too - I'll be good with pkgdown & building site when the time comes. If you're really keen though, feel free to make a move on it!

**Comment by jpcompartir** [comment]
Latest on the roadmap is that we're working towards having pipelines, datasets (user uploaded datasets not tf datasets/ the datasets library quite yet) and a predict function working + documentation and vignettes made for these things. The good news is that we're nearly done. We need to tidy up some documentation, add 1 or 2 more vignettes & then test the functionality beyond what's covered by the unit tests.

It's also possible to load tokenizers, load models (varying degrees of specificity e.g. just an AutoModel or AutoModelForSequenceClassification etc.), use sentence transformers to embed documents, use the inference API, and a host of other things that we probably won't communicate initially to keep things trim - but experienced users of the transformers library should find they can do a lot in the initial release if they go digging.

Any bugs, pain points etc. anyone has come across would be great to know now so that we can work on fixing them ahead of the first release.
